src/utilities/extendSchema.js

```diff
@@ -230,6 +230,8 @@ export function extendSchema(
     types,
     directives: getMergedDirectives(),
     astNode: schema.astNode,
+    allowedLegacyNames:
+      schema.__allowedLegacyNames && schema.__allowedLegacyNames.slice(0),
```

Comment:
You don't need to use `slice` since it's defined as:
https://github.com/graphql/graphql-js/blob/b333b30b238d040d6a8cf959d54f037ee26ae2c6/src/type/schema.js#L86

So `allowedLegacyNames: schema.__allowedLegacyNames` should be enough.

Reply:
This is why I needed the equality check, basically. Flow didn’t want to accept a read-only array for the config, which is typed to accept a mutable array: https://github.com/graphql/graphql-js/blob/b333b30b238d040d6a8cf959d54f037ee26ae2c6/src/type/schema.js#L248

I was surprised too, though, so not being very familiar with Flow I’ll do another sanity check.